### ISSUE TITLE
[feature] Maintenance-mode network configs

### ISSUE THREAD

**Issue by attilaolah** [siderolabs/omni#103]
### Problem Description

We have a number of Talos nodes that require non-default network configuration to be able to connect to Omni (bond + vlan). Currently we add these as kernel parameters to the Talos ISO, which allows us to provision these machines.

However, once a machine is "destroyed" and goes back to maintenance modes, these kernel parameters seem to disappear, and the machines lose connectivity. The only way to reset them seems to be using the "Reset Talos installation" option from the boot media.

### Solution

It would be nice if we could provide a maintenance mode config that would include, ideally, the network config we want.

### Alternative Solutions

If having maintenance mode network config is not feasible, an alternate solution would be for machines to go back to using the kernel configs from the original install media, when the machines are "destroyed". This would preserve the kernel parameters so the machines could reconnect.

**Comment by smira** [comment]
This depends on https://github.com/siderolabs/talos/issues/10925, which will be part of Talos 1.12 by the end of the year.

**Comment by carlosjgp** [comment]
I'm interested on this too

Provisioning Talos OS from a customer network which doesn't allow certain DNSs

**Comment by smira** [comment]
This should be now available now I believe via embedded machine config + Talos 1.12+ network config + Omni preserving this config automatically. @Unix4ever ?

**Comment by utkuozdemir** [comment]
Yes, since Omni 1.9, embedded machine configs are preserved and maintenance mode config patches are supported. From 1.9 changelog:
> Machine Config Patches in Maintenance Mode
Omni can now apply machine-level config patches while a machine is still in maintenance mode, not just after it joins a cluster. The patches go on top of the configuration the machine already runs, next to the SideroLink documents Omni manages, and Omni will not apply a document that installs Talos and pulls the machine out of maintenance. Omni also keeps whatever configuration a machine connects with as a low-priority, user-owned patch. So a machine that arrives with its own config (say a TrustedRootsConfig document) keeps it, and your own patches still win.

So I'm closing this as done.
